File: testbed_ruby/test/associations/unidirectional_many_test.rb
require 'test/unit'
require './src-gen-umple/student_p'
require './src-gen-umple/mentor_p'
require './src-gen-umple/program_p'

module CruiseTest
class UnidirectionalManyTest < Test::Unit::TestCase

  def test_add_student
    m = MentorP.new("blah")
    s = StudentP.new(1)
    m.add_student(s)
    assert_equal(s,m.get_student(0))
  end
  
  def test_addSameStudent
    m = MentorP.new("blah")
    s = StudentP.new(1)
    
    assert_equal(true,m.add_student(s))

    assert_equal(1,m.number_of_students)
    assert_equal(s,m.get_student(0))
  end  
  
  def test_remove_student
    m = MentorP.new("blah")
    s = StudentP.new(1)
    
    assert_equal(true,m.add_student(s))
    assert_equal(true,m.remove_student(s))
    assert_equal(false,m.remove_student(s))

    assert_equal(0,m.number_of_students)
  end   


  def test_addMultipleMentors
    m = MentorP.new("blah")
    m2 = MentorP.new("blah")

    s = StudentP.new(99)
    assert_equal(0,m.number_of_students)
    assert_equal(0,m2.number_of_students)

    m.add_student(s)
    
    assert_equal(1,m.number_of_students)
    assert_equal(0,m2.number_of_students)

    m2.add_student(s)

    assert_equal(1,m.number_of_students)
    assert_equal(1,m2.number_of_students)

    assert_equal(s,m.get_student(0))
    assert_equal(s,m2.get_student(0))
  end
  
  def test_deleteDoesNotRemoveStudentsFromMentors
    m = MentorP.new("blah")
    m2 = MentorP.new("blah")
    m3 = MentorP.new("blah")
    
    s = StudentP.new(99)
    s2 = StudentP.new(98)
    s3 = StudentP.new(97)
    
    m.add_student(s)
    m.add_student(s2)
    
    m2.add_student(s)
    m3.add_student(s3)
    
    s.delete
    
    assert_equal(2,m.number_of_students)
    assert_equal(1,m2.number_of_students)
    assert_equal(1,m3.number_of_students)
  end
  
end
end

File: testbed_php/test/associations/UnidirectionalManyTest.php
<?php

class UnidirectionalManyTest extends UnitTestCase
{

  public function test_addStudent()
  {
    $m = new MentorP("blah");
    $s = new StudentP(1);
    $m->addStudent($s);
    $this->assertEqual($s,$m->getStudent_index(0));
  }
  

  public function test_addSameStudent()
  {
    $m = new MentorP("blah");
    $s = new StudentP(1);
    
    $this->assertEqual(true,$m->addStudent($s));

    $this->assertEqual(1,$m->numberOfStudents());
    $this->assertEqual($s,$m->getStudent_index(0));
  }  
  

  public function test_removeStudent()
  {
    $m = new MentorP("blah");
    $s = new StudentP(1);
    
    $this->assertEqual(true,$m->addStudent($s));
    $this->assertEqual(true,$m->removeStudent($s));
    $this->assertEqual(false,$m->removeStudent($s));

    $this->assertEqual(0,$m->numberOfStudents());
  }   


  public function test_addMultipleMentors()
  {
    $m = new MentorP("blah");
    $m2 = new MentorP("blah");

    $s = new StudentP(99);
    $this->assertEqual(0,$m->numberOfStudents());
    $this->assertEqual(0,$m2->numberOfStudents());

    $m->addStudent($s);
    
    $this->assertEqual(1,$m->numberOfStudents());
    $this->assertEqual(0,$m2->numberOfStudents());

    $m2->addStudent($s);

    $this->assertEqual(1,$m->numberOfStudents());
    $this->assertEqual(1,$m2->numberOfStudents());

    $this->assertEqual($s,$m->getStudent_index(0));
    $this->assertEqual($s,$m2->getStudent_index(0));
  }
  

  public function test_deleteDoesNotRemoveStudentsFromMentors()
  {
    $m = new MentorP("blah");
    $m2 = new MentorP("blah");
    $m3 = new MentorP("blah");
    
    $s = new StudentP(99);
    $s2 = new StudentP(98);
    $s3 = new StudentP(97);
    
    $m->addStudent($s);
    $m->addStudent($s2);
    
    $m2->addStudent($s);
    $m3->addStudent($s3);
    
    $s->delete();
    
    $this->assertEqual(2,$m->numberOfStudents());
    $this->assertEqual(1,$m2->numberOfStudents());
    $this->assertEqual(1,$m3->numberOfStudents());
  }
  
}
?>
